### PR TITLE
[clang/cas] Switch `-fmodule-file-cache-key` option to being `MultiArg`

### PR DESCRIPTION
**Pull request by akyrtzi**
Previously parsing for the option assumes that `=` cannot appear in the CASID string, which isn’t a good assumption because `=` is a common character in `base64` strings. Switch to `MultiArg` option style which eliminates the issue of what string separator to use.

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci Please test

File: clang/lib/Frontend/CompilerInvocation.cpp [hunk 1]
```diff
@@ -664,6 +664,16 @@ static void GenerateArg(SmallVectorImpl<const char *> &Args,
                     Opt.getKind(), 0, Value);
 }
 
+static void GenerateMultiArg(SmallVectorImpl<const char *> &Args,
+                             llvm::opt::OptSpecifier OptSpecifier,
+                             ArrayRef<StringRef> Values,
+                             CompilerInvocation::StringAllocator SA) {
+  Option Opt = getDriverOptTable().getOption(OptSpecifier);
+  Args.push_back(SA(Opt.getPrefix() + Opt.getName()));
```
Comment (benlangmuir) [inline clang/lib/Frontend/CompilerInvocation.cpp:674]: The single-arg case seems to call `denormalizeString`; I'm wondering if that should happen here too? Maybe @jansvoboda11 could review this bit

File: clang/test/CAS/fmodule-file-cache-key-errors.c [hunk 2]
```diff
@@ -16,19 +16,19 @@
 // RUN:   -fcache-compile-job -Rcompile-job-cache &> %t/invalid.txt
 // RUN: cat %t/invalid.txt | FileCheck %s -check-prefix=INVALID
 
-// INVALID: error: option '-fmodule-file-cache-key' should be of the form <path>=<key>
+// INVALID: error: unknown argument: '-fmodule-file-cache-key=INVALID'
```
Comment (benlangmuir) [inline clang/test/CAS/fmodule-file-cache-key-errors.c:19]: Could we add a test here for the wrong number of arguments?

File: clang/lib/Frontend/CompilerInvocation.cpp (hunk 1 above)
Comment (akyrtzi) [reply clang/lib/Frontend/CompilerInvocation.cpp:674]: AFAICT `denormalize` is for doing the right thing based on the option class (e.g. is it joined?)
Here we know the specific option class but it's missing an assertion check, I'll add that.

Comment (akyrtzi) [reply clang/lib/Frontend/CompilerInvocation.cpp:674]: > it's missing an assertion check, I'll add that.

Done.

File: clang/test/CAS/fmodule-file-cache-key-errors.c (hunk 2 above)
Comment (akyrtzi) [reply clang/test/CAS/fmodule-file-cache-key-errors.c:19]: Done.

**Comment by akyrtzi** [comment]
@swift-ci Please test

**Review by benlangmuir — APPROVED** [review]

**Pull request merged**